noir-projects/aztec-nr/aztec/src/macros/functions/call_interface_stubs.nr

```diff
@@ -0,0 +1,243 @@
+use crate::macros::utils::{
+    add_to_field_slice, AsStrQuote, compute_fn_selector, is_fn_private, is_fn_view,
+};
+use std::meta::{type_of, unquote};
+
+comptime global FROM_FIELD: TypedExpr = {
+    let from_field_trait = quote { protocol_types::traits::FromField }.as_trait_constraint();
+    let function_selector_typ =
+        quote { protocol_types::abis::function_selector::FunctionSelector }.as_type();
+    function_selector_typ.get_trait_impl(from_field_trait).unwrap().methods().filter(|m| {
+        m.name() == quote { from_field }
+    })[0]
+        .as_typed_expr()
+};
+
+comptime global SERIALIZED_ARGS_SLICE_NAME: Quoted = quote { serialized_args };
+
+pub comptime fn stub_fn(f: FunctionDefinition) -> Quoted {
+    let is_static_call = is_fn_view(f);
+    let is_void = f.return_type() == type_of(());
+
+    if is_fn_private(f) {
```

Comment:
This is the main change here. I branch into individual cases and there I return the complete quote from there.